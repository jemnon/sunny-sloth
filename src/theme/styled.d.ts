import 'styled-components';
import {
  Breakpoints,
  Colors,
  ColorMode,
  FocusColors,
  Fonts,
  FontSizes,
  Header,
  HoverColors,
  MaxWidths,
  Spacing,
  ZIndex,
} from '../types/theme';

declare module 'styled-components' {
  export interface DefaultTheme {
    breakpoints: Breakpoints;
    colors: Colors;
    focusColors: FocusColors;
    fonts: Fonts;
    fontSizes: FontSizes;
    header: Header;
    hoverColors: HoverColors;
    maxWidths: MaxWidths;
    spacing: Spacing;
    transition: string;
    zIndex: ZIndex;
  }
}
