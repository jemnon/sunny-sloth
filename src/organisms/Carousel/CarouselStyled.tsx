import styled from 'styled-components';
import { up } from 'styled-breakpoints';
import { ButtonReset } from '../../atoms/Button';
import {
  Easing,
  CarouselButtonProps,
  CarouselContentProps,
  CarouselDotProps,
} from './carousel-types';

export const CarouselContainer = styled.section<{ carouselWidth?: string }>`
  position: relative;
  width: ${({ carouselWidth: width = '100%' }): string => width};
`;

export const CarouselContent = styled.div<CarouselContentProps>`
  display: flex;

  position: relative;
  z-index: 1;

  width: ${({ w: width }): string => `${width}px`};
  height: 100%;

  transition: all 0.25s ${({ easing = 'ease-in-out' }): Easing => easing};
  transform: ${({ xPos = 0 }): string => `translate3d(${xPos}px, 0, 0)`};
`;

export const CarouselBody = styled.div`
  position: relative;
  z-index: 0;

  width: 100%;
  height: 100%;

  overflow: hidden;
`;

export const CarouselNav = styled.nav`
  position: relative;
  z-index: 2;

  display: flex;
  align-items: center;
  justify-content: center;

  & + * {
    margin-top: ${({ theme: { spacing } }): string => spacing.sm4};
  }
`;

export const CarouselList = styled.ul`
  display: flex;
  align-items: center;
`;

export const CarouselListItem = styled.li`
  margin-left: 0.25rem;
  margin-right: 0.25rem;
`;

export const CarouselDot = styled.button<CarouselDotProps>`
  ${ButtonReset};

  border-radius: 100%;
  width: 0.5rem;
  height: 0.5rem;

  cursor: pointer;

  background-color: ${({ isCurrent }): string =>
    `rgba(255, 255, 255, ${isCurrent ? '100' : '0.5'})`};
`;

export const CarouselButton = styled.button<CarouselButtonProps>`
  ${ButtonReset};

  display: flex;
  justify-content: center;
  align-items: center;

  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  z-index: 2;
  ${({ direction = 'left' }): string =>
    direction === 'left' ? `left: 0.75rem` : `right: 0.75rem`};

  width: 2rem;
  height: 2rem;

  ${up('md')} {
    width: 3rem;
    height: 3rem;
  }

  background-color: rgba(0, 0, 0, 0.4);

  opacity: ${({ isDisabled }): string => (isDisabled ? '0.30' : '100%')};
  cursor: ${({ isDisabled }): string =>
    isDisabled ? 'not-allowed' : 'pointer'};
  border-radius: 100%;
`;

export const CarouselControls = styled.section`
  position: absolute;
  bottom: ${({ theme: { spacing } }): string => spacing.sm4};
  left: 50%;
  transform: translateX(-50%);

  padding-left: ${({ theme: { spacing } }): string => spacing.md4};
  padding-right: ${({ theme: { spacing } }): string => spacing.md4};
`;
