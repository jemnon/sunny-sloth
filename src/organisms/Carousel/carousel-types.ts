export type Easing = 'ease' | 'ease-in-out' | 'ease-in' | 'ease-out' | 'linear';

export interface CarouselContentProps {
  easing?: Easing;
  transitionDuration?: string;
  xPos?: number;
  h?: number;
  w?: number;
}

export interface CarouselDotProps {
  isCurrent: boolean;
}

type Direction = 'left' | 'right';

export interface CarouselButtonProps {
  direction?: Direction;
  isDisabled?: boolean;
}
