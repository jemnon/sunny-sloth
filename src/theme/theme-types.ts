export interface Breakpoints {
  sm: string;
  md: string;
  lg: string;
  xlg: string;
}

export interface Fonts {
  lato: string;
  latoBold: string;
  latoItalic: string;
  noto: string;
  notoBold: string;
}

export interface Colors {
  cream: string;
  darkTeal: string;
  teal: string;
  lightBrown: string;
  nearBlack: string;
  nearWhite: string;
  darkOrange: string;
  orange: string;
  white: string;
  red: string;
  green: string;
}

export type ColorMode = 'dark' | 'light';

export interface MaxWidths {
  sm: string;
  md: string;
  lg: string;
}

export interface Spacing {
  sm1: string;
  sm2: string;
  sm3: string;
  sm4: string;
  md1: string;
  md2: string;
  md3: string;
  md4: string;
  lg1: string;
  lg2: string;
  lg3: string;
  lg4: string;
}

export interface ZIndex {
  z0: string;
  z1: string;
  z2: string;
  z3: string;
  z4: string;
  z5: string;
  z6: string;
  z999: string;
  z9999: string;
}
