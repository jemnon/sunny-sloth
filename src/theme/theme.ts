const theme = {
  breakpoints: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
    xlg: '1440px',
  },
  fonts: {
    latoBold:
      `"Lato-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato-Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    notoBold:
      `"Noto Serif TC-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
  },
  spacing: {
    sm1: '0.25rem' /* 4px */,
    sm2: '0.5rem' /*  8px */,
    sm3: '0.75rem' /* 12px */,
    sm4: '1rem' /* 16px */,
    md1: '1.25rem' /* 20px */,
    md2: '1.5rem' /* 24px */,
    md3: '1.75rem' /* 28px */,
    md4: '2rem' /* 32px */,
    lg1: '2.25rem' /* 36px */,
    lg2: '2.5rem' /* 40px */,
    lg3: '2.75rem' /* 44px */,
    lg4: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
