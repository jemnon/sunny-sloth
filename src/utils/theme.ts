import { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
  breakpoints: {
    /* TODO: remove once conversion to styled-breakpoints is done */
    desktop: 'screen and (min-width: 960px)',
    tablet: 'screen and (min-width: 480px) and (max-width: 959px)',
    mobile: 'screen and (max-width: 479px)',
    /* ^^ */
    sm: '480px',
    md: '960px',
    lg: '1440px',
    xlg: '1920px',
  },
  fonts: {
    latoBold:
      `"Lato Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1440px',
  },
  spacing: {
    s1: '0.25rem' /* 4px */,
    s2: '0.5rem' /*  8px */,
    s3: '0.75rem' /* 12px */,
    s4: '1rem' /* 16px */,
    s5: '1.25rem' /* 20px */,
    s6: '1.5rem' /* 24px */,
    s7: '1.75rem' /* 28px */,
    s8: '2rem' /* 32px */,
    s9: '2.25rem' /* 36px */,
    s10: '2.5rem' /* 40px */,
    s11: '2.75rem' /* 44px */,
    s12: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
