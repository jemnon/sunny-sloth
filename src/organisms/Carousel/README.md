# Carousel

### Description
Component description

```js
import Carousel from '@fenderdigital/ui/FILL IN TYPECarousel';
```

### Props
* `propName` - property name and description 
