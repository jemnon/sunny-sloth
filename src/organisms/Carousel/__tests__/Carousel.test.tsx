import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Carousel from '../Carousel';
import theme from '../../../theme';

describe('<Carousel />', (): void => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Carousel>
        <div style={{ width: '100px', height: '200px' }} />
        <div style={{ width: '100px', height: '200px' }} />
        <div style={{ width: '100px', height: '200px' }} />
        <div style={{ width: '100px', height: '200px' }} />
      </Carousel>
    </ThemeProvider>,
  );
  it('exists', (): void => {
    expect(Carousel).toBeDefined();
  });
  it('renders defaults correctly', (): void => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
