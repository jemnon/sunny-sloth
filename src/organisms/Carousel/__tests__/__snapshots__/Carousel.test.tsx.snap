// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> renders defaults correctly 1`] = `
<section
  className="sc-fFubgz bQgpDU"
>
  <div
    className="sc-idOhPF iaUvbu"
  >
    <div
      className="sc-bkzZxe hVtwZE"
    >
      <div
        style={
          Object {
            "height": "200px",
            "width": "100px",
          }
        }
      />
      <div
        style={
          Object {
            "height": "200px",
            "width": "100px",
          }
        }
      />
      <div
        style={
          Object {
            "height": "200px",
            "width": "100px",
          }
        }
      />
      <div
        style={
          Object {
            "height": "200px",
            "width": "100px",
          }
        }
      />
    </div>
    <button
      className="sc-fKFyDc jIAyRB"
      direction="left"
      onClick={[Function]}
    >
      <svg
        className="sc-gKsewC dFuSAH"
        fill="#fff"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "marginRight": "-3px",
          }
        }
        viewBox="0 0 9.9 16.97"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.48 0L1.41 7.07 0 8.48l8.48 8.49 1.42-1.41-7.07-7.08L9.9 1.41 8.48 0z"
          fillRule="evenodd"
        />
      </svg>
    </button>
    <button
      className="sc-fKFyDc cWEiZc"
      direction="right"
      onClick={[Function]}
    >
      <svg
        className="sc-kEjbxe fRRqum"
        fill="#fff"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "marginLeft": "3px",
          }
        }
        viewBox="0 0 9.9 16.97"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.41 0L0 1.41l7.07 7.07L0 15.56l1.41 1.41L8.48 9.9 9.9 8.48 1.41 0z"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </div>
  <section
    className="sc-bBXqnf crXpW"
  >
    <nav
      className="sc-dIUggk iLkYPn"
    >
      <ul
        className="sc-hHftDr kRIGKa"
      >
        <li
          className="sc-dmlrTW dRbkSR"
        >
          <button
            className="sc-kfzAmx fOpycm"
            onClick={[Function]}
          />
        </li>
        <li
          className="sc-dmlrTW dRbkSR"
        >
          <button
            className="sc-kfzAmx ePJKBY"
            onClick={[Function]}
          />
        </li>
        <li
          className="sc-dmlrTW dRbkSR"
        >
          <button
            className="sc-kfzAmx ePJKBY"
            onClick={[Function]}
          />
        </li>
        <li
          className="sc-dmlrTW dRbkSR"
        >
          <button
            className="sc-kfzAmx ePJKBY"
            onClick={[Function]}
          />
        </li>
      </ul>
    </nav>
  </section>
</section>
`;
